src/compiler/checker.ts

```diff
@@ -10702,8 +10783,18 @@ namespace ts {
                 }
 
                 let derived = getTargetSymbol(getPropertyOfObjectType(type, base.name));
-                if (derived) {
-                    let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+                let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+
+                if (!derived) { // derived class inherits base without override/redeclaration
+                    let derivedClassDecl = getDeclarationOfKind(type.symbol, SyntaxKind.ClassDeclaration);
+                    Debug.assert(derivedClassDecl !== undefined);
+
+                    // It is an error to inherit an abstract member without implementing it or being declared abstract.
+                    if ((baseDeclarationFlags & NodeFlags.Abstract) && !(derivedClassDecl.flags & NodeFlags.Abstract)) {
+                        error(derivedClassDecl, Diagnostics.Non_abstract_class_0_does_not_implement_inherited_abstract_member_1_2,
+                            typeToString(type), typeToString(baseType), symbolToString(baseProperty));
+                    }
+                } else { // derived !== undefined -- derived overrides base

```

Comment:
This code was broken. It's been fixed and needs to have a different test.

I put the new block at the top because it is shorter, so you can see where the block ends in less than one screen. I thought this would aid readability.